internal/venice-common/src/main/java/com/linkedin/venice/controllerapi/ControllerClient.java

```diff
@@ -1156,6 +1156,21 @@ public MultiStoreInfoResponse getClusterStores(String clusterName) {
     return request(ControllerRoute.GET_STORES_IN_CLUSTER, params, MultiStoreInfoResponse.class);
   }
 
+  /**
+   * This method gets a list of store names that are ready for compaction.
+   * @param clusterName, the name of the cluster to query for compaction eligible stores
+   * @return The list of store names that are ready for compaction.
+   */
+  public MultiStoreInfoResponse getStoresForCompaction(String clusterName) {
+    QueryParams params = newParams().add(CLUSTER, clusterName);
+    return request(ControllerRoute.GET_STORES_FOR_COMPACTION, params, MultiStoreInfoResponse.class);
+  }
+
+  public ControllerResponse triggerRepush(String storeName) {
+    QueryParams params = newParams().add(NAME, storeName);
+    return request(ControllerRoute.COMPACT_STORE, params, ControllerResponse.class);
+  }
```

Comment:
I think I follow but I'd like to make sure. Let's sync on this to get clarity.

Reply:
Yes. I was suggesting the same and add a comment like
`TODO: Use byte[] to pass parameters instead of QueryParams as it is a post method so that you don't forget when you work later on this in the next PR`

Reply:
- comment added
- parked `ControllerClient::triggerRepush` for later PR